backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/ir/Ir.kt

```diff
@@ -89,6 +89,13 @@ internal class KonanIr(context: Context, irModule: IrModuleFragment): Ir<Context
         return originalModuleIndex.classes[descriptor]
                 ?: symbols.symbolTable.referenceClass(descriptor).owner
     }
+
+    fun getPropertyGetterByName(descriptor: ClassDescriptor, name: Name): IrSimpleFunctionSymbol {
+        val property = descriptor.unsubstitutedMemberScope
+                .getContributedVariables(name, NoLookupLocation.FROM_BACKEND)
+                .single()
+        return context.ir.symbols.symbolTable.referenceSimpleFunction(property.getter!!)
```

Comment:
This looks incorrect to me. Have you tried writing link test on this optimization (i.e. when enum is declared in one module and lowered `when` is written in another module)?

Reply:
Nope and lowering fails even earlier on this line :)
https://github.com/JetBrains/kotlin-native/blob/7e5c1713d9ac91945d08d3e09241ecf30e4390b6/backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/lower/EnumClassLowering.kt#L192